gammapy/utils/cluster.py

```diff
@@ -0,0 +1,51 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Utils for hierarchical/agglomerative clustering."""
+
+import scipy.cluster.hierarchy as sch
+
+
+def hierarchical_clustering(
+    features, linkage_kwargs=None, fcluster_kwargs=None, standard_scaler=False
+):
+    """Hierarchical clustering using given features.
+
+    Parameters
+    ----------
+    features : array
+        (N x M) array with N observations and M features.
+    linkage_kwargs : dict
+        Arguments forwarded to `scipy.cluster.hierarchy.linkage`
+    fcluster_kwargs : dict
+        Arguments forwarded to `scipy.cluster.hierarchy.fcluster`
+    standard_scaler : bool
+        Standardize features by removing the mean and scaling to unit variance.
+        Default is False.
+
+    Returns
+    -------
+    ind_clusters : array
+        Array of cluster ID
+    features : array
+        Scaled features
+    """
+
+    if standard_scaler:
```

Comment:
Maybe rather add this option to where the table is generated? This has the advantage of being able to look at the values for diagnosis...

Reply:
I also added a standard_scaler function so the scaled features can be computed separately